### PR TITLE
Update data docs

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of docs revamp #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @justusschock @awaelchli @ninginthecloud

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: docs/source/guides/data.rst [hunk 1]
```diff
@@ -277,17 +282,17 @@ When using PackedSequence, do 2 things:
         return x, y
 
 
-    # In module
+    # In LightningModule
     def training_step(self, batch, batch_nb):
```
Comment (akihironitta) [inline docs/source/guides/data.rst:286]: nit: for consitency
```suggestion
    def training_step(self, batch, batch_idx):
```

**Review by akihironitta — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/guides/data.rst [hunk 2]
```diff
@@ -344,9 +348,10 @@ option when using sequential data.
         seq_data = ["A", "long", "time", "ago", "in", "a", "galaxy", "far", "far", "away"]
         iterable_dataset = CustomDataset(seq_data)
 
-        dataloader = DataLoader(dataset=iterable_dataset, batch_size=5)
+        dataloader = DataLoader(data=iterable_dataset, batch_size=5)
```
Comment (awaelchli) [inline docs/source/guides/data.rst:351]: copy paste error?

File: docs/source/guides/data.rst [hunk 3]
```diff
@@ -277,17 +282,17 @@ When using PackedSequence, do 2 things:
...
         y = rnn.pack_sequence(batch[1], enforce_sorted=False)
 
-----------
 
 Truncated Backpropagation Through Time (TBPTT)
 ==============================================
+
 There are times when multiple backwards passes are needed for each batch.
-For example, it may save memory to use Truncated Backpropagation Through Time when training RNNs.
+For example, it may save memory to use ``Truncated Backpropagation Through Time`` when training RNNs.
```
Comment (awaelchli) [inline docs/source/guides/data.rst:295]:
```suggestion
For example, it may save memory to use **Truncated Backpropagation Through Time** when training RNNs.
```
Was the intention to make it bold?

File: docs/source/guides/data.rst [hunk 4]
```diff
@@ -230,15 +226,21 @@ needs to wrap the DataLoaders with `CombinedLoader`.
...
-set is not available at the time your model was declared. Simply pass the test set to the :func:`~pytorch_lightning.trainer.trainer.Trainer.test` method:
+    def validation_step(self, batch, batch_idx):
+        batch_a = batch["a"]
+        batch_b = batch["b"]
+
+
+Evaluate with Additional DataLoaders
+====================================
+
+You can evaluate your models using additonal dataloaders even if the dataloader specific hooks hasn't been defined within your
```
Comment (awaelchli) [inline docs/source/guides/data.rst:241]:
```suggestion
You can evaluate your models using additonal dataloaders even if the dataloader specific hooks haven't been defined within your
```

File: docs/source/guides/data.rst [hunk 5]
```diff
@@ -230,15 +226,21 @@ needs to wrap the DataLoaders with `CombinedLoader`.
...
+:class:`~pytorch_lightning.core.lightning.LightningModule` instance. For example, this would be the case if your test data
```
Comment (awaelchli) [inline docs/source/guides/data.rst:242]:
```suggestion
:class:`~pytorch_lightning.core.lightning.LightningModule`. For example, this would be the case if your test data
```

The definition of hooks/methods is in the class

File: docs/source/guides/data.rst [hunk 6]
```diff
@@ -197,19 +184,22 @@ Furthermore, Lightning also supports nested lists and dicts (or a combination).
...
-=================================
-For validation and test DataLoaders, you can pass a single DataLoader or a list of them. This optional named
+Multiple Validation/Test/Predict DataLoaders
+============================================
+
+For validation, test and predict DataLoaders, you can pass a single DataLoader or a list of them. This optional named
 parameter can be used in conjunction with any of the above use cases. You can choose to pass
 the batches sequentially or simultaneously, as is done for the training step.
-The default mode for validation and test DataLoaders is sequential.
+The default mode for these DataLoaders is sequential. Note that when using sequential a sequence of dataloaders you need
```
Comment (awaelchli) [inline docs/source/guides/data.rst:194]: "when using sequential a sequence of "

File: docs/source/guides/data.rst [hunk 7]
```diff
@@ -197,19 +184,22 @@ Furthermore, Lightning also supports nested lists and dicts (or a combination).
...
+to add an additional argument ``dataloader_idx`` in their corresponding step specific hook. The corresponding loop will process
+the dataloaders in sequential order i.e. first dataloader will be processed first completely, then the second one, and so on.
```
Comment (awaelchli) [inline docs/source/guides/data.rst:196]:
```suggestion
the dataloaders in sequential order, i.e., the first dataloader will be processed completely, then the second one, and so on.
```

File: docs/source/guides/data.rst [hunk 8]
```diff
@@ -218,8 +208,14 @@ See the following for more details for the default sequential option:
...
-To combine batches of multiple test and validation DataLoaders simultaneously, one
-needs to wrap the DataLoaders with `CombinedLoader`.
+
+    def validation_step(self, batch, batch_idx, dataloader_idx):
+        ...
+
+
+To combine batches of multiple DataLoaders simultaneously, one
+needs to wrap the DataLoaders with :class:`~pytorch_lightning.trainer.supporters.CombinedLoader`.
+Here you don't need to add any additional ``dataloader_idx`` argument here.
```
Comment (awaelchli) [inline docs/source/guides/data.rst:218]:
```suggestion
Here you don't need to add any additional ``dataloader_idx`` argument.
```

File: docs/source/guides/data.rst [hunk 9]
```diff
@@ -36,19 +27,21 @@ There are a few different data containers used in Lightning:
    * - :class:`~torch.utils.data.IterableDataset`
      - The PyTorch :class:`~torch.utils.data.IterableDataset` represents a stream of data.
    * - :class:`~torch.utils.data.DataLoader`
-     - The PyTorch :class:`~torch.utils.data.DataLoader` represents a Python iterable over a DataSet.
+     - The PyTorch :class:`~torch.utils.data.DataLoader` represents a Python iterable over a Dataset.
    * - :class:`~pytorch_lightning.core.datamodule.LightningDataModule`
-     -  A :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is simply a collection of: a training DataLoader, validation DataLoader(s), test DataLoader(s) and predict DataLoader(s), along with the matching transforms and data processing/downloads steps required.
+     -  A :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is simply a collection of: train DataLoader(s), validation DataLoader(s), test DataLoader(s) and predict DataLoader(s), along with the matching transforms and data processing/downloads steps required.
```
Comment (awaelchli) [inline docs/source/guides/data.rst:32]:
```suggestion
     -  A :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is simply a collection of: training DataLoader(s), validation DataLoader(s), test DataLoader(s) and predict DataLoader(s), along with the matching transforms and data processing/downloads steps required.
```

**Review by awaelchli — APPROVED** [review]

File: docs/source/guides/data.rst (hunk 3 above)
Comment (rohitgr7) [reply docs/source/guides/data.rst:295]: nah! but it's better.

File: docs/source/guides/data.rst (hunk 2 above)
Comment (rohitgr7) [reply docs/source/guides/data.rst:351]: oops! need to be more careful.
